scripts/o3de/o3de/project_manager_interface.py

```diff
@@ -0,0 +1,285 @@
+#
+# Copyright (c) Contributors to the Open 3D Engine Project.
+# For complete copyright and license terms please see the LICENSE at the root of this distribution.
+#
+# SPDX-License-Identifier: Apache-2.0 OR MIT
+#
+#
+"""
+Contains functions for the project manager to call that gather data from o3de scripts and massage it
+"""
+
+import logging
+import pathlib
+
+from o3de import cmake, disable_gem, download, enable_gem, engine_properties, engine_template, manifest, project_properties, register, repo
+
+logger = logging.getLogger('o3de.project_manager_interface')
+logging.basicConfig(format=utils.LOG_FORMAT)
+
+
+#### Engine methods ###
+
+def get_engine_info(engine_name: str = None, engine_path: str = None) -> dict or None:
+    """
+        If engine_path is provided then continue
+        Elif only engine_name is provided then call get_registered and use returned path as engine_path
+        Elif neither parameter is provided call get_this_engine_path and use returned path as engine_path
+
+        Call get_engine_json_data using engine_path to populate initial engine data
+
+        Call load_o3de_manifest or get default folders for gems, projects, ect. using get_o3de_..._folder
+        Insert either default folder from manifest or get_o3de_..._folder into engine dict for all folders
+
+        Call get_manifest_engines and check that engine is in the list if it is set registered attr to True
+
+        :param engine_name: Engine name to find engine path for and get json data for
+        :param engine_path: Engine path to gather json data for
+
+        :return dict containing engine info if it was found. Otherwise None is returned
+    """
+    return dict()
+
+
+def set_engine_info(engine_info: dict):
+    """
+        Call get_engine_info using path from engine_info
+
+        Call edit_engine_props if any relevant properties have changed
+
+        Call register if any relevant properties have changed
+
+        :param engine_info: dict containing values to change in engine
+    """
+    pass
+
+
+#### Template methods ###
+
+def get_project_template_infos() -> list:
+    """
+        Get list of registered project templates using get_templates_for_project_creation
+
+        Gather together list of dicts for each template using get_template_json_data
+
+        :return list of dicts containing template infos.
+    """
+    return list()
+
+
+#### Project methods ###
+
+def register_project(project_path: str, remove: bool = False):
```

Comment:
Just a thought since this is intended on being a abstraction layer, why not break out the register/unregister calls into separate functions for clarity?

Reply:
I went back and forth on this and went with this because the total number of methods was less only having one method, though I can see your idea being less confusing, so I'll change it.